### ISSUE TITLE
Update/patch (PUT/PATCH) strips "id" from saved object

### ISSUE THREAD

**Issue by joshuajabbour** [feathersjs/feathers-mongodb#25]
I'm still working on a test case, but I'm having issues with the update action. Not sure if it's in this module or another, but here are the details:

- Using custom `id` field (`id`, not `_id`)
- `PUT /resources/123` with body `{"id": "123", "title": "#123"}` strips out the id field upon save.

This doesn't help, as it's not honored either; the `id` field _is_ added to `hook.data`, but stripped again sometime before saving:

```js
  service
    .before({
      update (hook) {
        // Copy ID from request into object.
        hook.data.id = hook.id
      }
  })
```

**Comment by joshuajabbour** [comment]
The same problem _does not_ occur when `PATCH` is used, even when the `id` field is provided. However, if the `id` is altered in a `PATCH` request, it is not actually changed (this is probably desired behavior).

The problem with the behavior in the `PUT` request is that the `id` is ultimately stripped, so the resource then becomes an orphan without an `id`.

**Comment by ekryski** [comment]
@joshuajabbour thanks for reporting! I released v2.1.1 that has your fix.

**Comment by joshuajabbour** [comment]
I've added comments on https://github.com/feathersjs/feathers-mongodb/commit/5980ed6873851727db374f5a19e8d50ed1a94352. While part the initial issue was fixed, I think it introduces new bugs.
